### PR TITLE
Guard against NULL in fix_invalid_vehicle_sprite_sizes()

### PR DESCRIPTION
**Pull request by rwjuk**

### CODE REVIEW

**Comment by janisozaur** [comment]
Was it one of the recent crashes?

**Comment by rwjuk** [comment]
@janisozaur No, this is just me being cautious!

File: src/openrct2/ride/ride.c [hunk 1]
```diff
@@ -8681,6 +8681,15 @@ void fix_invalid_vehicle_sprite_sizes()
             while (rideSpriteIndex != SPRITE_INDEX_NULL) {
                 rct_vehicle *vehicle = GET_VEHICLE(rideSpriteIndex);
                 rct_ride_entry_vehicle *vehicleEntry = vehicle_get_vehicle_entry(vehicle);
+
+                if (vehicle == NULL ||
+                    vehicle == (rct_vehicle*)-1 ||
```
Comment (IntelOrca) [inline src/openrct2/ride/ride.c]: vehicle can't be `-1`

File: src/openrct2/ride/ride.c [hunk 2]
```diff
@@ -8681,6 +8681,14 @@ void fix_invalid_vehicle_sprite_sizes()
             while (rideSpriteIndex != SPRITE_INDEX_NULL) {
                 rct_vehicle *vehicle = GET_VEHICLE(rideSpriteIndex);
                 rct_ride_entry_vehicle *vehicleEntry = vehicle_get_vehicle_entry(vehicle);
+
+                if (vehicle == NULL ||
+                    vehicleEntry == NULL ||
+                    vehicleEntry == (rct_ride_entry_vehicle*)-1)
+                {
+                    continue;
```
Comment (janisozaur) [inline src/openrct2/ride/ride.c]: Shouldn't this `break` instead? `rideSpriteIndex` is not getting updated in the `while` loop, so you would end up selecting the very same vehicle over and over again in an endless loop.

Comment (rwjuk) [inline src/openrct2/ride/ride.c]: Yes! Nice catch. Will fix later, feel free to do it yourself in the mean time if you want to get this merged.

**Pull request merged**